pkg/services/anonymous/anonimpl/api/api.go

```diff
@@ -0,0 +1,120 @@
+package api
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/anonymous/anonimpl/anonstore"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+const (
+	thirtyDays = 30 * 24 * time.Hour
+)
+
+type AnonDeviceServiceAPI struct {
+	cfg            *setting.Cfg
+	store          anonstore.AnonStore
+	accesscontrol  accesscontrol.AccessControl
+	RouterRegister routing.RouteRegister
+	log            log.Logger
+}
+
+func NewAnonDeviceServiceAPI(
+	cfg *setting.Cfg,
+	anonstore anonstore.AnonStore,
+	accesscontrol accesscontrol.AccessControl,
+	routerRegister routing.RouteRegister,
+) *AnonDeviceServiceAPI {
+	return &AnonDeviceServiceAPI{
+		cfg:            cfg,
+		store:          anonstore,
+		accesscontrol:  accesscontrol,
+		RouterRegister: routerRegister,
+		log:            log.New("anon.api"),
+	}
+}
+
+func (api *AnonDeviceServiceAPI) RegisterAPIEndpoints() {
+	auth := accesscontrol.Middleware(api.accesscontrol)
+	api.RouterRegister.Group("/api/anonymous", func(anonRoutes routing.RouteRegister) {
```

Comment:
this would make sense for the stats page, perhaps we could explore that. The endpoints are needed for the anonymous users tab in the `users` page. I also included the endpoints first under /api/admin/stats or closer to it but we decided to make these endpoints part of the registration of the service, together w @kalleep . any more thoughts for this @kalleep

Reply:
I'd keep it all all undef one endpoint - `/api/admin/stats` as it's much easier. For the Users page tab we could still use the `/api/admin/stats` endpoint and just get the relevant fields from there.

Reply:
Yeah that sounds reasonable

Reply:
@Jguer and I had a discussion, and we will reintroduce the anon service into the `hs` to call it from the `/api/admin/stats`. It is a real anti-pattern to call two endpoints for one component in the frontend.

Thx for the comments